lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescaleset.py

```diff
@@ -852,6 +927,24 @@ def exec_module(self, **kwargs):
 
                         vmss_resource.virtual_machine_profile.storage_profile.data_disks = data_disks
 
+                    # Before creating VM accept terms of plan if `accept_terms` is True
+                    if self.accept_terms is True:
+                        if not self.plan or not all([self.plan.get('name'), self.plan.get('product'), self.plan.get('publisher')]):
+                            self.fail("parameter error: plan must be specified and include name, product, and publisher")
```

Comment:
You can use `required_if` to make sure that `plan` must be specified if `accept_terms` is `True`.

Reply:
I think this needs to be the other way around.  If a plan is specified then accept_terms must be True. I left it in for now until I learn the syntax for required_if.

Reply:
Your logic from the code above is: if `accept_terms` is `True`, `plan` must be specified. (That can also be modelled by `required_if`.) If you want the logic the other way around, you have to change your code (and you can't use `required_if` AFAIK).